lms/djangoapps/verify_student/models.py

```diff
@@ -881,3 +882,48 @@ def send_request(self):
         log.debug("Return message:\n\n{}\n\n".format(response.text))
 
         return response
+
+
+class VerificationCheckpoint(models.Model):
+    """represents a point at which a user is challenged to reverify his or her identity.
+        Each checkpoint is uniquely identified by a (course_id, checkpoint_name) tuple.
+    """
+
+    CHECKPOINT_CHOICES = (
+        ("midterm", "midterm"),
+        ("final", "final"),
+    )
+
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    checkpoint_name = models.CharField(unique=True, max_length=32, choices=CHECKPOINT_CHOICES)
```

Comment:
# Adding unique constraint on 'VerificationCheckpoint', fields ['course_id', 'checkpoint_name']
 db.create_unique('verify_student_verificationcheckpoint', ['course_id', 'checkpoint_name'])
The index is already there for these two fields in migration file